crates/astria-core/src/protocol/transaction/v1alpha1/mod.rs

```diff
@@ -381,35 +417,50 @@ enum UnsignedTransactionErrorKind {
         raw::UnsignedTransaction::type_url()
     )]
     DecodeAny(#[source] prost::DecodeError),
+    #[error("`actions` field does not make a valid `ActionGroup`")]
+    ActionGroup(#[source] action_group::Error),
 }
 
-pub struct TransactionParamsBuilder<TChainId = std::borrow::Cow<'static, str>> {
+pub struct UnsignedTransactionBuilder<TChainId = std::borrow::Cow<'static, str>> {
```

Comment:
Having a generic parameter seems a little excessive to me.  I think it'd be fine to just have `chain_id: String` and have the setter as `pub fn chain_id<T: Into<String>>(self, chain_id: T)`